### PR TITLE
Switch to system installed Boost on Linux.

### PR DESCRIPTION
**Pull request by PetrosKataras**
This PR switches the default behavior on Linux from linking with the precompiled Boost libs to system installed versions.

Requires `libboost-filesystem-dev` to be installed through the package manager of your platform/distribution.

This should resolves issues like [this](https://discourse.libcinder.org/t/linker-errors-mongodb-cxx-driver-on-linux/504), #1915, #1968  and a few more of similar nature.

### CODE REVIEW

File: proj/cmake/platform_linux.cmake [hunk 1]
```diff
@@ -191,17 +191,9 @@ if( GSTREAMER_GL_INCLUDE_DIRS AND GSTREAMER_GL_LIBRARIES )
...
-else()
-	set( LINUX_LIB_DIRECTORY "${CINDER_PATH}/lib/linux/${CINDER_ARCH}/" )
-	list( APPEND CINDER_LIBS_DEPENDS
-		${LINUX_LIB_DIRECTORY}/libboost_system.a
-		${LINUX_LIB_DIRECTORY}/libboost_filesystem.a
-	)
-endif()
+find_package( Boost 1.54 REQUIRED COMPONENTS system filesystem )
+list( APPEND CINDER_LIBS_DEPENDS ${Boost_LIBRARIES} )
+list( APPEND CINDER_INCLUDE_SYSTEM_PRIVATE ${Boost_INCLUDE_DIRS} )
```
Comment (richardeakin) [inline proj/cmake/platform_linux.cmake]: Does this mean that we no longer be shipping any boost libraries on linux, in other words the `CINDER_BOOST_USE_SYSTEM` option is the only option? If so I'm not against it, just thought we had discussed switching the default behavior but still shipping prebuilt binaries. Maybe that has shown itself to be more trouble than its worth, though.

Comment (PetrosKataras) [reply proj/cmake/platform_linux.cmake]: My opinion would be to make the default and only behavior systemwide installed Boost. It has proven indeed problematic in the past even by just checking out the Boost submodule and then having mixed includes ( e.g. coming from Cinder's version and from a system wide installed version of Boost ) Not linking with the prebuilt binaries is not really enough in this case.

**Comment by PetrosKataras** [comment]
Had forgotten that the binaries are not part of the submodule, so I just removed them and this should be complete now from my side I believe.

**Comment by rdtor** [comment]
Hi Petros how can i use this PR, do i need to build cinder again from your repo https://github.com/PetrosKataras/Cinder/tree/linux-boost-use-system-install?

**Comment by PetrosKataras** [comment]
The simplest thing to try first would be to remove your previous Cinder build folder and re-compile Cinder with:

`cd Cinder && mkdir build && cd build && cmake .. -DCINDER_BOOST_USE_SYSTEM=1 && make -j4`

Make sure you have `libboost-filesystem-dev` installed and also make sure that you completely remove the `Cinder/include/boost` folder before trying this out.

If you want to apply this PR then you should do what @gaborpapp suggested in #2088 in order to checkout a branch locally in your machine with the changes applied and then recompile Cinder as usual. 

i.e `git fetch origin pull/2007/head:pr2007`

**Comment by rdtor** [comment]
I applied this PR and followed what @gaborpapp suggested and the build was succesfull thanks

**Comment by andrewfb** [comment]
I'm intending to merge this in the next day or two unless anyone has any counterpoints

**Comment by richardeakin** [comment]
@PetrosKataras question:

> The simplest thing to try first would be to remove your previous Cinder build folder and re-compile Cinder with:
> 
> cd Cinder && mkdir build && cd build && cmake .. -DCINDER_BOOST_USE_SYSTEM=1 && make -j4
> 

Doesn't this PR remove the use of `-DCINDER_BOOST_USE_SYSTEM=1`? As in, after this PR the only option is to use system boost on linux.

**Comment by PetrosKataras** [comment]
@richardeakin yeah this is correct - the part of the comment you quote was in the case that the user didn't want to apply this PR locally and instead try installed boost with the current Cinder setup ( i.e without this PR merged ).

**Comment by PetrosKataras** [comment]
Just noticed the README mentioning a recursive git checkout. This should not be required for Linux actually after this PR is merged and in fact it should be discouraged since this would pull the include files for Boost which could still cause some weird clashes with system installed Boost ( ..it would also pull TinderBox which is also completely irrelevant to the Linux setup.. ).

I suppose we should add a note about this?

**Comment by richardeakin** [comment]
Just tested this with both the command line and Visual Studio Code on Ubuntu 19.04, looking good!

If it can cause header mismatches, I do think we should remove the `--recursive` flag from the git command for the linux docs pages. We may end up using other submodules later, but hopefully we'll have ditched the boost submodule by that point.

**Pull request merged**